### ISSUE TITLE
Priority-based relayer coordination protocol: required modifications

### ISSUE THREAD

**Issue by svyatonik** [paritytech/parity-bridges-common#2618]
Filing this issue to track remaining work that needs to be done **after** #2605 is merged:
- [ ] relayer can deregister from lane without any penalty only after it has delivered at least one message (so we assume it'll be working for at least one epoch). Another option to deregister is to be pushed out by relayer with better bid (lower expected reward). See https://github.com/paritytech/parity-bridges-common/pull/2605#discussion_r1352615442 for details;
- [ ] randomly select block that ends relayers "auction". For BH - use randomness coming from relay chain. See https://github.com/paritytech/parity-bridges-common/pull/2605#discussion_r1352615442 for details;
- [ ] to avoid contention during single auction, maybe we need to limit number of times when relayer is able to change his bid - e.g. to 1 time per auction (or use some alternate solution). This is only possible if we'll implement previous point, though. If we won't do that, and if there's a lot of relayers, figting for the slot, we may end up with many unneeded transactions where relayers are registering using a slightly better price (`worst - 1`).
